### PR TITLE
hw3

### PR DESCRIPTION
**Pull request by IgorMuchnik**

### CODE REVIEW

File: src/main/java/hw3/components/AbstractComponent.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package hw3.components;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+public class AbstractComponent {
```
Comment (DmitryKhodakovsky) [inline src/main/java/hw3/components/AbstractComponent.java]: In the name of the class you have mentioned that it `Abstract`. However you missed `abstract` modifier for class

File: src/main/java/hw3/components/AbstractComponent.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+  protected WebDriver driver;
+  protected WebDriverWait wait;
+
+  public AbstractComponent(WebDriver driver) {
```
Comment (DmitryKhodakovsky) [inline src/main/java/hw3/components/AbstractComponent.java]: Why do you set constructor as `public`

File: src/main/java/hw3/components/HeaderMenu.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  }
+
+  public boolean isHeaderMenuItemsDisplayed(){
+    waitForHeaderMenuElementsToBeVisible();
+    boolean isDisplayed = false;
+    for (WebElement element : headerMenuElements) {
+      isDisplayed = element.isDisplayed();
+    }
+    return isDisplayed;
+  }
```
Comment (DmitryKhodakovsky) [inline src/main/java/hw3/components/HeaderMenu.java]: I see similar method second time. I guess it code duplication and could be extracted to the base page class

File: src/main/java/hw3/components/LogSection.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+  public boolean isDropdownLogDisplayed(String dropdownValue) {
+    waitForLogsToBeVisible();
+    boolean isDisplayed = false;
+    for (WebElement log : logs) {
+      if (log.getText().contains("Colors: value changed to " + dropdownValue)) {
+        isDisplayed = true;
+      }
+    }
+    return isDisplayed;
```
Comment (DmitryKhodakovsky) [inline src/main/java/hw3/components/LogSection.java]: These methods are code duplication. They have similar logic, that could be combined in the one method

File: src/main/java/hw3/components/SideBarMenu.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    super(driver);
+  }
+
+  public boolean isSidebarElementsDisplayed(){
+    waitForSidebarMenuToBeVisible();
+    boolean isDisplayed = false;
+    for (WebElement element : sideBarMenuElements) {
+      isDisplayed = element.isDisplayed();
+    }
+    return isDisplayed;
```
Comment (DmitryKhodakovsky) [inline src/main/java/hw3/components/SideBarMenu.java]: code duplication

File: src/main/java/hw3/pages/AbstractPage.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
+package hw3.pages;
+
+import hw3.components.HeaderMenu;
+import hw3.components.SideBarMenu;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+public class AbstractPage {
```
Comment (DmitryKhodakovsky) [inline src/main/java/hw3/pages/AbstractPage.java]: Why did you miss `abstract` modified?

File: src/main/java/hw3/pages/AbstractPage.java [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+  protected WebDriver driver;
+  protected SideBarMenu sideBarMenu;
+  protected HeaderMenu headerMenu;
+  protected WebDriverWait wait;
+
+  public AbstractPage(WebDriver driver) {
```
Comment (DmitryKhodakovsky) [inline src/main/java/hw3/pages/AbstractPage.java]: Why is it `public`?

File: src/main/java/hw3/pages/SimpleElementsPage.java [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+    }
+  }
+
+  public void setRadio(String checkboxName) {
+    wait.until(ExpectedConditions.visibilityOfAllElements(radioElements));
+    for (WebElement checkBoxElement : radioElements) {
+      if(!checkBoxElement.isSelected() & checkBoxElement.getText().equals(checkboxName))
+        checkBoxElement.click();
+    }
+  }
```
Comment (DmitryKhodakovsky) [inline src/main/java/hw3/pages/SimpleElementsPage.java]: Code duplication. 

`setRadio(String checkboxName)` - radio or checkbox?

**Review by DmitryKhodakovsky — APPROVED** [review]
HW3 is accepted with comments. could be merged

File: src/main/java/hw3/components/AbstractComponent.java (hunk 1 above)
Comment (IgorMuchnik) [reply src/main/java/hw3/components/AbstractComponent.java]: fixed. forgot to write

File: src/main/java/hw3/components/AbstractComponent.java (hunk 2 above)
Comment (IgorMuchnik) [reply src/main/java/hw3/components/AbstractComponent.java]: fixed. it is more correct to do with protected constructor

File: src/main/java/hw3/components/HeaderMenu.java (hunk 3 above)
Comment (IgorMuchnik) [reply src/main/java/hw3/components/HeaderMenu.java]: refactored

File: src/main/java/hw3/components/LogSection.java (hunk 4 above)
Comment (IgorMuchnik) [reply src/main/java/hw3/components/LogSection.java]: refactored

File: src/main/java/hw3/components/SideBarMenu.java (hunk 5 above)
Comment (IgorMuchnik) [reply src/main/java/hw3/components/SideBarMenu.java]: refactored

File: src/main/java/hw3/pages/AbstractPage.java (hunk 6 above)
Comment (IgorMuchnik) [reply src/main/java/hw3/pages/AbstractPage.java]: fixed

File: src/main/java/hw3/pages/AbstractPage.java (hunk 7 above)
Comment (IgorMuchnik) [reply src/main/java/hw3/pages/AbstractPage.java]: fixed. it is more correct to do with protected constructor

File: src/main/java/hw3/pages/SimpleElementsPage.java (hunk 8 above)
Comment (IgorMuchnik) [reply src/main/java/hw3/pages/SimpleElementsPage.java]: code was refactored, fixed the misspell

**Pull request merged**